datacube/utils/xarray_geoextensions.py

```diff
@@ -32,8 +32,21 @@ def _xarray_affine(obj):
 
     dims = crs.dimensions
 
-    xres, xoff = data_resolution_and_offset(obj[dims[1]].values)
-    yres, yoff = data_resolution_and_offset(obj[dims[0]].values)
+    xres = data_resolution(obj[dims[1]].values)
+    yres = data_resolution(obj[dims[0]].values)
+    if xres is None and yres is None:
+        try:
+            xres, yres = obj.resolution
```

Comment:
I don't quite understand. You are checking for this attribute here, but you don't seem to have attached it anywhere.